src/types/typeGuards/single.ts

```diff
@@ -0,0 +1,813 @@
+import ts from "typescript";
+
+import { isTypeFlagSet } from "../../flags";
+import { type FreshableIntrinsicType } from "./compound";
+
+/**
+ * A "any" intrinsic type.
+ *
+ * @category Type Types
+ */
+export interface AnyType extends IntrinsicType {
```

Comment:
[Naming] A name like _"any type"_ sounds like something more wide than _"type"_. How about...

```suggestion
export interface IntrinsicAnyType extends IntrinsicType {
```

...with the same rename on the rest of the intrinsic types?

Reply:
What about `BooleanLiteralType` and other types that don't directly extend `IntrinsicType`?

(`BooleanLiteralType` extends `UnknownLiteralType` which extends `FreshableIntrinsicType` which extends `IntrinsicType`)

Reply:
I don't think the "literal" types need renaming, but if these "intrinsic" were to follow the same naming scheme, they would be called `AnyIntrinsicType`. However, that still has the same sort of issue as just `AnyType`.

Reply:
Yeah I think `BooleanLiteralType` is fine. The _`Literal`_ part of that carries weight.